### PR TITLE
update(cni): update from v1.5.3 -> v1.6.0

### PR DESCRIPTION
**Pull request by metral**
### Proposed changes

- update(cni): update from v1.5.3 -> v1.6.0
Based on the AWS CNI v1.6.0 YAML manifest: https://git.io/Jvla1
CNI config variables: https://git.io/Jvlah

### Related issues (optional)

Closes https://github.com/pulumi/pulumi-eks/issues/330

### CODE REVIEW

File: nodejs/eks/cni.ts [hunk 1]
```diff
@@ -86,6 +86,24 @@ export interface VpcCniOptions {
...
+    vethPrefix?: pulumi.Input<string>;
+
+    /**
+     * Used to configure the MTU size for attached ENIs. The valid range is
+     * from 576 to 9001.
+     *
+     * Defaults to 9001.
+     */
+    eniMtu?: pulumi.Input<number>;
 }
```
Comment (lblackstone) [inline nodejs/eks/cni.ts]: I don't see the following configs that are present in the upstream docs:
- ENI_CONFIG_ANNOTATION_DEF
- ENI_CONFIG_LABEL_DEF
- AWS_VPC_K8S_CNI_RANDOMIZESNAT
- AWS_VPC_K8S_CNI_EXCLUDE_SNAT_CIDRS
- MINIMUM_IP_TARGET
- MAX_ENI
- INTROSPECTION_BIND_ADDRESS
- DISABLE_INTROSPECTION
- DISABLE_METRICS
- ADDITIONAL_ENI_TAGS
- CLUSTER_NAME

Is that intentional?

File: nodejs/eks/cni.ts [hunk 2]
```diff
@@ -131,6 +151,16 @@ function computeVpcCniYaml(cniYamlText: string, args: VpcCniInputs): string {
     } else {
         env.push({name: "AWS_VPC_K8S_CNI_LOG_FILE", value: "stdout"});
     }
+    if (args.vethPrefix) {
+        env.push({name: "AWS_VPC_K8S_CNI_VETHPREFIX", value: args.vethPrefix.toString()});
+    } else {
+        env.push({name: "AWS_VPC_K8S_CNI_VETHPREFIX", value: "eni"});
```
Comment (lblackstone) [inline nodejs/eks/cni.ts]: Do you actually need to set this? It should already default to this value, right?

File: nodejs/eks/cni.ts [hunk 3]
```diff
@@ -131,6 +151,16 @@ function computeVpcCniYaml(cniYamlText: string, args: VpcCniInputs): string {
...
+    }
+    if (args.eniMtu) {
+        env.push({name: "AWS_VPC_ENI_MTU", value: args.eniMtu.toString()});
+    } else {
+        env.push({name: "AWS_VPC_ENI_MTU", value: "9001"});
```
Comment (lblackstone) [inline nodejs/eks/cni.ts]: Same here.

File: nodejs/eks/cni.ts (hunk 1 above)
Comment (metral) [reply nodejs/eks/cni.ts]: It's possible that these envvars did not exist at the initial time `cni.ts` was written. We've also have not gotten user demand to add them in.

Adding them in is straight forward, but I'm not sure if we want to include all envvars now and those forthcoming in the future, and I'm now reevaluating what the stance should be.

cc @lukehoban ^

File: nodejs/eks/cni.ts (hunk 2 above)
Comment (metral) [reply nodejs/eks/cni.ts]: > It should already default to this value

Right. Though, historically, we've set defaults for envvars that are configurable in pulumi/eks to aid the user in understanding top-level args, and bubbling them up in docs.

File: nodejs/eks/cni.ts (hunk 1 above)
Comment (metral) [reply nodejs/eks/cni.ts]: Opened an issue to track missing vars: https://github.com/pulumi/pulumi-eks/issues/337

File: nodejs/eks/cni.ts (hunk 2 above)
Comment (metral) [reply nodejs/eks/cni.ts]: Going to continue with this as its the same pattern we currently use. Will revisit envvars in https://github.com/pulumi/pulumi-eks/issues/337

File: nodejs/eks/cni.ts (hunk 3 above)
Comment (metral) [reply nodejs/eks/cni.ts]: Same as https://github.com/pulumi/pulumi-eks/pull/331#discussion_r382787035

**Pull request merged**